Patch to correct CMakeLists.txt so external TBB will be used instead of internal build.
Patch to convert innan to std::isnan
J. Sassmannshausen / GSTT
diff --git a/usher-0.4.1.orig/CMakeLists.txt b/usher-0.4.1/CMakeLists.txt
index 94754a8..f919b34 100644
--- a/usher-0.4.1.orig/CMakeLists.txt
+++ b/usher-0.4.1/CMakeLists.txt
@@ -35,9 +35,19 @@ endif()
 
 include_directories(${Protobuf_INCLUDE_DIRS})
 
-include(${TBB_DIR}/cmake/TBBBuild.cmake)
-tbb_build(TBB_ROOT ${TBB_DIR} CONFIG_DIR TBB_DIR MAKE_ARGS tbb_cpf=1)
-find_package(TBB REQUIRED tbbmalloc tbbmalloc_proxy tbb_preview)
+find_library(TBB_LIB_PATH tbb)
+if (TBB_LIB_PATH)
+    find_path(TBB_INCLUDE_PATH tbb)
+    link_libraries(${TBB_LIB_PATH})
+    message(STATUS "TBB lib found in ${TBB_LIB_PATH}")
+    find_library(TBBMALLOC_LIB_PATH tbbmalloc)
+    link_libraries(${TBBMALLOC_LIB_PATH})
+    message(STATUS "TBBMALLOC lib found in ${TBBMALLOC_LIB_PATH}")
+    find_library(TBBMALLOC_PROXY_LIB_PATH tbbmalloc_proxy)
+    link_libraries(${TBBMALLOC_PROXY_LIB_PATH})
+    message(STATUS "TBBMALLOC_PROXY lib found in ${TBBMALLOC_PROXY_LIB_PATH}")
+    include_directories(${TBB_INCLUDE_PATH})
+endif()
 
 FIND_PACKAGE(Boost COMPONENTS program_options iostreams filesystem date_time REQUIRED)
 
diff --git a/usher-0.4.1.orig/src/matUtils/introduce.cpp b/usher-0.4.1/src/matUtils/introduce.cpp
index 9b31efb..6e206d2 100644
--- a/usher-0.4.1.orig/src/matUtils/introduce.cpp
+++ b/usher-0.4.1/src/matUtils/introduce.cpp
@@ -357,7 +357,7 @@ std::map<std::string, float> get_assignments(MAT::Tree* T, std::unordered_set<st
                     float vir = (static_cast<float>(min_to_in) / static_cast<float>(in_leaves.size()));
                     float r = (vir/vor);
                     float c = (1/(1+r));
-                    if (isnan(c)) {
+                    if (std::isnan(c)) {
                         fprintf(stderr, "ERROR: Invalid introduction assignment calculation. Debug information follows.\n");
                         fprintf(stderr, "min %ld, mout %ld, ols %ld, ils %ld,", min_to_in, min_to_out, out_leaves.size(), in_leaves.size());
                         fprintf(stderr, " vor %f, vir %f, r %f\n", vor, vir, r);
